common/metrics/events.go

```diff
@@ -78,67 +81,68 @@ func MetricHandlerFromConfig(logger log.Logger, c *Config) MetricHandler {
 	)
 }
 
-// NewEventMetricProvider provides an eventMetricProvider given event.Exporter struct
-func NewEventMetricProvider(h MetricHandler) *eventMetricProvider {
+// NewEventsMetricProvider provides an eventsMetricProvider given event.Exporter struct
+func NewEventsMetricProvider(h MetricHandler) eventsMetricProvider {
 	eo := &event.ExporterOptions{
-		DisableLogging: true,
-		DisableTracing: true,
+		DisableLogging:   true,
+		DisableTracing:   true,
+		EnableNamespaces: false,
 	}
 
-	return &eventMetricProvider{
+	return eventsMetricProvider{
 		context: event.WithExporter(context.Background(), event.NewExporter(h, eo)),
 		tags:    []Tag{},
 	}
 }
 
 // WithTags creates a new MetricProvder with provided []Tag
 // Tags are merged with registered Tags from the source MetricProvider
-func (emp *eventMetricProvider) WithTags(tags ...Tag) MetricProvider {
-	return &eventMetricProvider{
+func (emp eventsMetricProvider) WithTags(tags ...Tag) MetricProvider {
+	t := append(make([]Tag, 0, len(emp.tags)+len(tags)), emp.tags...)
```

Comment:
no need to make a new slice, append handle that automatically, simply do newTags := append(t.tags, tags...)

Reply:
This was part of my data race troubleshooting, this slice copy might have been the correct fix.

I will confirm

Reply:
This is a race condition if I do not make a new slice.

w/o adding a fence, what is a better way to avoid the data race?

Reply:
I don't understand the race condition here. And I don't understand the needs to change method from pointer to struct either.

Reply:
Yeah, It is not entirely clear but there are clear concurrency issues that present writes from multiple go routines and panic during tests.